### PR TITLE
create events endpoints for student

### PR DESCRIPTION
**Pull request by justinkang01**
### Summary 

This PR deals with the interactions between the student and event data variables, as well as creating common functions.

- [x] Modified types.ts to allow students to have a list of events they are attending
- [x ] Modified /student/sid to properly retrieve events list
- [x] Created /student/sid/events/ to retrieve all events a student is attending
- [x] Created /student/sid/events/eid to add an event to the list of events sid is attending

### Test Plan 

Hard to test this on multiple users, and currently no events.ts to create events, so used firestore console to create temporary events and made sure the get requests worked.

### Breaking Changes  

<!-- - Database schema change (anything that changes DynamoDB document structure)
<!-- - Other change that could cause problems (Detailed in notes)

### CODE REVIEW

File: backend/router/student.ts [hunk 1]
```diff
@@ -48,39 +41,65 @@ router.get('/', async (_, res) => {
...
+    );
+    const eventsAttending = await getDataFromPromises(
+      student,
+      'events',
+      'EventType'
+    );
+    const studentInstance = {
+      id: student.id,
+      firstName: student.firstName,
+      lastName: student.lastNsame,
```
Comment (20austinw) [inline backend/router/student.ts]: Is this a typo here?

**Review by 20austinw — DISMISSED** [review]
I think there is a small typo in studentInstance. Otherwise, the code looks good to merge!

File: backend/router/student.ts (hunk 1 above)
Comment (justinkang01) [reply backend/router/student.ts]: shoot good eye, ill fix it real quick.

**Review by JustKong13 — APPROVED** [review]
Everything looks good to me!

**Pull request merged**